src/nl/rubensten/texifyidea/util/Labels.kt

```diff
@@ -7,67 +7,84 @@ import nl.rubensten.texifyidea.index.BibtexIdIndex
 import nl.rubensten.texifyidea.index.LatexCommandsIndex
 import nl.rubensten.texifyidea.psi.BibtexId
 import nl.rubensten.texifyidea.psi.LatexCommands
+import nl.rubensten.texifyidea.settings.TexifySettings
 
 /**
  * Finds all the defined labels in the fileset of the file.
  *
  * @return A set containing all labels that are defined in the fileset of the given file.
  */
-fun PsiFile.findLabelsInFileSet(): Set<String> {
-    val latex = findLatexLabelsInFileSet()
-    val bibtex = findBibtexLabelsInFileSet()
-    return (latex + bibtex).toSet()
-}
+fun PsiFile.findLabelsInFileSet(): Set<String> = (findAllLabelsInFileSet() + findBibtexLabelsInFileSet()).toSet()
 
 /**
  * Finds all the defined latex labels in the fileset of the file.
  *
  * @return A set containing all labels that are defined in the fileset of the given file.
  */
-fun PsiFile.findLatexLabelsInFileSet(): Sequence<String> = LatexCommandsIndex.getItemsInFileSet(this)
-        .asSequence()
-        .filter { "\\label" == it.name || "\\bibitem" == it.name }
-        .mapNotNull(LatexCommands::getRequiredParameters)
-        .filter { it.isNotEmpty() }
-        .map { it.first() }
+fun PsiFile.findAllLabelsInFileSet(): Sequence<String> {
+    val commands = TexifySettings.getInstance().labelCommands
+    return findLabelingCommandsSequence()
+            .map { it.name to it.requiredParameters }
+            .filter { it.second.isNotEmpty() }
+            .mapNotNull {
+                val position = commands[it.first]?.position
+                if (position != null && it.second.size >= position)
+                    it.second[position - 1] else null
+            }.filterNot { it == "" }
+}
 
 /**
  * Finds all the defined bibtex labels in the fileset of the file.
  *
  * @return A set containing all labels that are defined in the fileset of the given file.
  */
-fun PsiFile.findBibtexLabelsInFileSet(): Sequence<String> = BibtexIdIndex.getIndexedIdsInFileSet(this)
-        .asSequence()
-        .map { it.text.substringEnd(1) }
+//fun PsiFile.findBibtexLabelsInFileSet(): Sequence<String> = BibtexIdIndex.getIndexedIdsInFileSet(this)
+//        .asSequence()
+//        .map { it.text.substringEnd(1) }
```

Comment:
Please remove commented out code.

Reply:
fixed in commit 4fc09f4f03cfda54b13164fb3b1d6e31c694f878